sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java

```diff
@@ -1717,4 +1724,61 @@ public void analyzeTasksEmptyInput(HttpClient httpClient, TextAnalyticsServiceVe
+        client = getTextAnalyticsAsyncClient(httpClient, serviceVersion);
+        analyzeBatchActionsPartialCompletedRunner(
+            (documents, tasks) -> {
+                SyncPoller<AnalyzeBatchActionsOperationDetail, PagedFlux<AnalyzeBatchActionsResult>> syncPoller =
+                    client.beginAnalyzeBatchActions(documents, tasks,
+                        new AnalyzeBatchActionsOptions().setIncludeStatistics(false)).getSyncPoller();
+                syncPoller.waitForCompletion();
+                PagedFlux<AnalyzeBatchActionsResult> result = syncPoller.getFinalResult();
+                validateAnalyzeBatchActionsResultList(false,
+                    Arrays.asList(getExpectedAnalyzeBatchActionsResult(
+                        IterableStream.of(Collections.emptyList()),
+                        IterableStream.of(asList(
+                            getExpectedRecognizePiiEntitiesActionResult(true, TIME_NOW, null,
+                                getActionError(INVALID_REQUEST, PII_TASK, "0")),
+                            getExpectedRecognizePiiEntitiesActionResult(
+                                false, TIME_NOW, getRecognizePiiEntitiesResultCollection(), null))),
+                        IterableStream.of(asList(
+                            getExpectedExtractKeyPhrasesActionResult(
+                                false, TIME_NOW, getExtractKeyPhrasesResultCollection(), null))))),
+                    result.toStream().collect(Collectors.toList()));
+            }
+        );
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.textanalytics.TestUtils#getTestParameters")
+    public void analyzeBatchActionsAllFailed(HttpClient httpClient, TextAnalyticsServiceVersion serviceVersion) {
+        client = getTextAnalyticsAsyncClient(httpClient, serviceVersion);
+        analyzeBatchActionsAllFailedRunner(
+            (documents, tasks) -> {
+                SyncPoller<AnalyzeBatchActionsOperationDetail, PagedFlux<AnalyzeBatchActionsResult>> syncPoller =
+                    client.beginAnalyzeBatchActions(documents, tasks,
+                        new AnalyzeBatchActionsOptions().setIncludeStatistics(false)).getSyncPoller();
+                syncPoller.waitForCompletion();
+                PagedFlux<AnalyzeBatchActionsResult> result = syncPoller.getFinalResult();
+
+                validateAnalyzeBatchActionsResultList(false,
+                    Arrays.asList(getExpectedAnalyzeBatchActionsResult(
+                        IterableStream.of(asList(
+                            getExpectedRecognizeEntitiesActionResult(true, TIME_NOW, null,
+                                getActionError(INVALID_REQUEST, ENTITY_TASK, "0")))),
+                        IterableStream.of(asList(
+                            getExpectedRecognizePiiEntitiesActionResult(true, TIME_NOW, null,
```

Comment:
@mssfang sorry I've been trying to parse the java code, but can't read it well enough :'(. Are the errors inserted into the main getter, i.e. if the second PII action result is an error, would getting the PII action results mean [GoodResponse, BadRespone]? Or are they returned in a separate getter `getActionError`.

Reply:
Yes,  The response will be [GoodResponse, BadResponse] that maps to the input action order. Each actionResult could be successful action result or action error.